packages/client/src/base/command-stack.ts

```diff
@@ -13,22 +13,38 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-import { injectable, multiInject, optional } from 'inversify';
-import { CommandStack, ICommand, SModelRoot, SetModelCommand, TYPES, UpdateModelCommand, distinctAdd, remove } from '~glsp-sprotty';
+import { injectable, multiInject, optional, preDestroy } from 'inversify';
+import { CommandStack, Disposable, Emitter, Event, ICommand, SModelRoot, SetModelCommand, TYPES, UpdateModelCommand } from '~glsp-sprotty';
 
 /**
  * A hook to listen for model root changes. Will be called after a server update
  * has been processed
  */
-export interface SModelRootListener {
+export interface ISModelRootListener {
```

Comment:
I'm just slightly wondering whether we should consider a name like `IGModelRootListener` for this since we are already breaking API and https://github.com/eclipse-glsp/glsp/issues/865 is something that will also be done at some point. It is really the only one of the types that should be affected by this.

Reply:
Wouldn't i make more sense to this as part of https://github.com/eclipse-glsp/glsp/issues/865?
Seems strange to me to use `GModelRoot` right now since the type does not exist.